app/jobs/save_instagram_photos_job.rb

```diff
@@ -1,16 +1,32 @@
 class SaveInstagramPhotosJob < ApplicationJob
   queue_as :default
+  attr_accessor :user, :instagram_photos
 
   def perform(user)
-    user&.instagram_photos&.each do |photo|
-      Photo.create(
-        user: user,
-        ig_id: photo['id'],
-        ig_media_url: photo['media_url'],
-        ig_permalink: photo['permalink'],
-        ig_caption: photo['caption'],
-        ig_timestamp: photo['timestamp']
-      )
+    @user = user
+    @instagram_photos = user&.instagram_photos
+    instagram_photos&.each do |photo|
+      create_photo(photo)
+      broadcast_photo_count
     end
   end
+
+  def create_photo(photo)
+    Photo.create(
+      user: user,
+      ig_id: photo['id'],
+      ig_media_url: photo['media_url'],
+      ig_permalink: photo['permalink'],
+      ig_caption: photo['caption'],
+      ig_timestamp: photo['timestamp']
+    )
+  end
+
+  def broadcast_photo_count
```

Comment:
Haven't been able to validate any of this stuff yet until I can figure out the connection setup piece.